test/e2e/nodemaintenance_test.go

```diff
@@ -170,8 +164,8 @@ func nodeMaintenanceTest(t *testing.T, f *framework.Framework, ctx *framework.Te
 		}
 		return true, nil
 	}); err != nil {
+		t.Logf("Failed to verify running phase: %v", err)
 		showDeploymentStatus(t, f)
-		t.Fatal(fmt.Errorf("Failed to verify running phase: %v", err))
```

Comment:
Surely these should not be merged?

Reply:
thats the e2e test; I needed it to stop  the test right at that place with an error message.they don't have a Fatalf method for that. Does that explain the change?

Reply:
showDeploymentStatus already calls Fatal, so it wouldn't continue to print t.Fatal(fmt.Errorf("Failed to verify running phase: %v", )). The change makes it easier to understand what went wrong during the e2e test.

Reply:
I think it's a mistake for `showDeploymentStatus()` to always fail the test.
Perhaps only do so if the caller passes in an error.

Reply:
done